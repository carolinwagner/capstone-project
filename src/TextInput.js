import React from 'react'
import styled from 'styled-components/macro'

const TextInput = ({ question, register, defaultValue }) => {
  const regex =
    question?.validationHookForm?.pattern &&
    new RegExp(question?.validationHookForm?.pattern)

  const validationClone = {
    ...question.validationHookForm,
    ...(regex && { pattern: regex }),
  }

  return (
    <StyledTextInput
      autoFocus
      id={question.name}
      type="text"
      placeholder={question.placeholder}
      name={question?.name || 'defaultTextInput'}
      defaultValue={defaultValue}
      ref={register(validationClone)}
      {...question?.validationNative}
    />
  )
}

export default TextInput

const StyledTextInput = styled.input`
  width: 100%;
  padding: 12px 20px;
  margin: 20px 0;
  display: inline-block;
  border: 1px solid var(--blue);
  border-radius: 4px;
`
