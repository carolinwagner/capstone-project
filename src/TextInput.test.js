import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import TextInput from './TextInput'

describe('TextInput', () => {
  it('renders TextInput correctly', () => {
    const tree = renderer.create(<TextInput />)
    expect(tree).toMatchSnapshot()
  })
})
