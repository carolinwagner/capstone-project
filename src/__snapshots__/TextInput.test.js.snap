// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders TextInput correctly 1`] = `undefined`;

exports[`TextInput renders correctly 1`] = `undefined`;
